packages/grafana-data/src/types/app.ts

```diff
@@ -64,13 +64,23 @@ export type AppPluginExtensionLinkConfig<C extends object = object> = {
   configure?: (extension: AppPluginExtensionLink, context?: C) => Partial<AppPluginExtensionLink> | undefined;
 };
 
+// A list of helpers that can be used in the command handler
+export type AppPluginExtensionCommandHelpers = {
+  // Opens a modal dialog and renders the provided React component inside it
+  // (It will always receive the `onDismiss()` prop for closing the modal manually), plus the additional `props` provided
+  openModal: <T extends object = object>(payload: {
+    component: React.ComponentType<{ onDismiss: () => void } & T>;
```

Comment:
> Make onDimisss optional.

That's a good point, will change 👍 

> Always unmount the component when dismissed. This way you also guarantee plugins don't create unclosable modals.

Hmm, the unmounting of the component is taken care of by the ModalManager (link below). The `onDismiss` is called by default when the user clicks on the close or outside of the modal, we only pass it down to the `openModal()` in case the user wants to dismiss the modal manually on a custom action. Did I misunderstand something?
https://github.com/grafana/grafana/blob/f8dc333ee468d2d38257b867724d26dcf621b0ac/public/app/core/services/ModalManager.ts#L46